telegram/_bot.py

```diff
@@ -4571,10 +4567,7 @@ async def send_invoice(
         if start_parameter is not None:
             data["start_parameter"] = start_parameter
         if provider_data is not None:
-            if isinstance(provider_data, str):
-                data["provider_data"] = provider_data
-            else:
-                data["provider_data"] = json.dumps(provider_data)
+            data["provider_data"] = provider_data
```

Comment:
if this can only be a string now for some reason, you would need to edit the doc string + type hints.

Reply:
it can be a string or an object which is json serializable, so its correct?

Reply:
then at least the doc is wrong, which says it would be encoded, and we should definitely note down the change in the doc anyway. But is there not a better type hint to say that an object needs to be json serializable?

Reply:
The encoding is done by the networking backend, same as for the other telegram objects that are accepted as parameter values. If that wasn't the case, the tests would fail :)

Reply:
So you are saying the same error would appear if I were to supply a non serializable object, just with a different traceback?

Reply:
yes. after all, everything that we pass to TG has to be JSON-dumped at some point. The new `BaseRequest` + `RequestData` just makes it more clear, where that's supposed to happen so that we no longer need it here.

Reply:
ah okay then